### PR TITLE
PR: Change setResizeMode to setSectionResizeMode

### LINKED ISSUE #3943: Qtpy 1.2.0 breaks Spyder with Qt4
## Description of your problem

Spyder crashes if I upgrade qtpy from 1.1.2 to 1.2.0

Originally I posted in the spyder-ide repository, but I was redirected here.
https://github.com/spyder-ide/qtpy/issues/95

**What steps will reproduce the problem?**

1. pip install -U qtpy (installs version 1.2.0 and breaks Spyder)
2. pip install qtpy==1.1.2 (everything working again) 

**What is the expected output? What do you see instead?**

Spyder doesn't start correctly, showing the following message:

```python-traceback
File "/home/developer/.virtualenvs/dev3/lib/python3.5/site-packages/spyder/app/mainwindow.py", line 3001, in main
mainwindow = run_spyder(app, options, args)
File "/home/developer/.virtualenvs/dev3/lib/python3.5/site-packages/spyder/app/mainwindow.py", line 2905, in run_spyder
main.setup()
File "/home/developer/.virtualenvs/dev3/lib/python3.5/site-packages/spyder/app/mainwindow.py", line 869, in setup
self.projects = Projects(self)
File "/home/developer/.virtualenvs/dev3/lib/python3.5/site-packages/spyder/plugins/projects.py", line 71, in init
self.setup_project(self.get_active_project_path())
File "/home/developer/.virtualenvs/dev3/lib/python3.5/site-packages/spyder/widgets/projects/explorer.py", line 226, in setup_project
show_all=self.show_all)
File "/home/developer/.virtualenvs/dev3/lib/python3.5/site-packages/spyder/widgets/explorer.py", line 195, in setup
self.common_actions = self.setup_common_actions()
File "/home/developer/.virtualenvs/dev3/lib/python3.5/site-packages/spyder/widgets/projects/explorer.py", line 55, in setup_common_actions
self.toggle_hscrollbar(self.show_hscrollbar)
File "/home/developer/.virtualenvs/dev3/lib/python3.5/site-packages/spyder/widgets/projects/explorer.py", line 70, in toggle_hscrollbar
self.header().setResizeMode(QHeaderView.ResizeToContents)
File "/home/developer/.virtualenvs/dev3/lib/python3.5/site-packages/qtpy/_patch/qheaderview.py", line 83, in setResizeMode
raise Exception('setResizeMode is only available in Qt4. Use '
Exception: setResizeMode is only available in Qt4. Use setSectionResizeMode instead.
```

## Versions and main components

* Spyder Version: 3.0.2
* Python Version: 3.5.2
* Operating system: Ubuntu 16.04
* Qt4 related info: Qt 4.8.7, PyQt 4.11.4, SIP 4.18.1

## Dependencies
jedi >=0.8.1     :  0.9.0 (OK)
matplotlib >=1.0 :  1.5.3 (OK)
nbconvert >=4.0  :  4.2.0 (OK)
numpy >=1.7      :  1.11.2 (OK)
pandas >=0.13.1  :  0.19.1 (OK)
pep8 >=0.6       :  1.7.0 (OK)
psutil >=0.3     :  5.0.0 (OK)
pyflakes >=0.6.0 :  1.3.0 (OK)
pygments >=2.0   :  2.1.3 (OK)
pylint >=0.25    :  1.6.4 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.4.8 (OK)
sympy >=0.7.3    :  1.0 (OK)

### PR DESCRIPTION
**Pull request by rlaverde**
Fixes #3943

-----------------

There is also another call to this method in https://github.com/spyder-ide/spyder/blob/master/spyder/widgets/variableexplorer/dataframeeditor.py#L432 but I tested and doesn't throw that error

### CODE REVIEW

**Comment by ccordoba12** [comment]
Thanks a lot for the quick response @rlaverde!

File: spyder/widgets/projects/explorer.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def toggle_hscrollbar(self, checked):
         if PYQT5:
             self.header().setSectionResizeMode(QHeaderView.ResizeToContents)
         else:
-            self.header().setResizeMode(QHeaderView.ResizeToContents)
```
Comment (ccordoba12) [inline spyder/widgets/projects/explorer.py]: We need a try/except here to support older `qtpy` versions. We could rewrite this `if` block as

```python
try:
    self.header().setSectionResizeMode(QHeaderView.ResizeToContents)
except:
    self.header().setResizeMode(QHeaderView.ResizeToContents)
```

and remove the `PYQT5` constant

**Comment by ccordoba12** [comment]
Please also change the other `setResizeMode` call even if it doesn't throw errors.

File: spyder/widgets/projects/explorer.py (hunk 1 above)
Comment (goanpeca) [inline spyder/widgets/projects/explorer.py]: Me like :+1:

**Comment by ccordoba12** [comment]
LGTM, thanks :-)

**Pull request merged**